hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -647,14 +661,16 @@ private void purgeScheduling(@NonNull final State state, final Instant then, fin
      * and no other side effects.
      *
      * @param parentTxn the user transaction to execute
-     * @param txnVersion the software version for the event containing the transaction
      * @param state the state to commit any direct changes against
+     * @param eventBirthRound the round in which the event was born
      * @return the stream output from executing the transaction
      */
     private HandleOutput executeSubmittedParent(
-            @NonNull final ParentTxn parentTxn, @NonNull final SemanticVersion txnVersion, @NonNull final State state) {
+            @NonNull final ParentTxn parentTxn, final long eventBirthRound, @NonNull final State state) {
         try {
-            if (isOlderSoftwareEvent(txnVersion)) {
+            final var platformStateStore =
+                    new ReadablePlatformStateStore(state.getReadableStates(PlatformStateService.NAME));
+            if (eventBirthRound <= platformStateStore.getLatestFreezeRound()) {
```

Comment:
It is still good to add this `eventBirthRound <= platformStateStore.getLatestFreezeRound()` to a method named `isOlderSoftwareEvent`. Why do we not need the `EVENT_STREAM_RECOVERY` condition here ?